File: jamf_api.rb
class JamfApi < BaseApi
  def initialize
    super("https://upstart.jamfcloud.com/JSSResource")
  end

  def get_users
    response = get("/users", base_options)
    response
  end

  def get_accounts
    response = get("/accounts", base_options)
    response
  end

  private

  def base_options
    options = {}
    options[:headers] = {
      accept: "application/xml",
      authorization: "Basic #{ENV["JAMF_BASE64"]}",
      "content-type" => "application/xml"
    }
    options
  end
end

File: jamf_api.py
import requests
import os


class JamfApi:
    def __init__(self):
        self.BASEURL = "https://upstart.jamfcloud.com/JSSResource"
        self.client = requests.Session()
        self.client.headers = {
                'accept':'application/json',
                'content-type':'application/json'
                }
        self.client.auth = (os.environ["JAMF_USERNAME"], os.environ["JAMF_PASSWORD"])

    def jamf_get(self, direct: bool, path, json=None, params=None):
        endpoint = ""
        if direct:
            endpoint = self.url(path)
        else:
            endpoint = path

        response = self.client.get(
            url=endpoint,
            json=json,
            params=params
        )

        return response

    def url(self, uri):
        return f"{self.BASEURL}{uri}"
